app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    storage = {}  # this will be stored in closure because it is nonlocal
```

Comment:
Please remove comments all over the project

Reply:
why should I remove the comments? It is good practice to comment on a code, isn`t it? What if I leave comments for myself?

Reply:
code should be self-descriptive. comments isn't a common practice in the Python community. 

if you want to describe how the function works, please, use docstrings